producer.go

```diff
@@ -112,10 +117,27 @@ func (p *Producer) Produce(message []byte, opts ...ProduceOpt) error {
 
 }
 
+func (opts *ProduceOpts) validateHeaderKey() error {
+	for i, _ := range opts.Header {
+		if strings.HasPrefix(i, "$memphis") {
+			return errors.New("Keys in headers should not start with $memphis")
+		}
+	}
+	return nil
+}
+
 // ProducerOpts.produce - produces a message into a station using a configuration struct.
 func (opts *ProduceOpts) produce(p *Producer) error {
+	memphisHeader := map[string][]string{"$memphisconnectionId": {p.conn.ConnId}, "$memphisproducedBy": {p.Name}}
+
+	err := opts.validateHeaderKey()
```

Comment:
not good this is an extra operation on the produce - really bad to implement it that way.
Headers should be a struct with a receiver function add(key, value), when this function called the validation should take place